cli/client.go

```diff
@@ -1355,6 +1357,90 @@ func (c *viamClient) machinesPartCopyFilesAction(
 	return nil
 }
 
+type robotsPartTunnelArgs struct {
+	Organization    string
+	Location        string
+	Machine         string
+	Part            string
+	LocalPort       int
+	DestinationPort int
+}
+
+// RobotsPartTunnelAction is the corresponding Action for 'machines part tunnel'.
+func RobotsPartTunnelAction(c *cli.Context, args robotsPartTunnelArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	return client.robotPartTunnel(c, args)
+}
+
+func tunnelTraffic(ctx *cli.Context, robotClient *client.RobotClient, local, dest int) error {
+	li, err := net.Listen("tcp", net.JoinHostPort("localhost", strconv.Itoa(local)))
+	if err != nil {
+		return fmt.Errorf("failed to create listener %w", err)
+	}
+	infof(ctx.App.Writer, "tunneling connections from local port %v to destination port %v on machine part...", local, dest)
+	defer func() {
+		if err := li.Close(); err != nil {
+			warningf(ctx.App.ErrWriter, "error closing listener: %s", err)
+		}
+	}()
+
+	var wg sync.WaitGroup
+	for {
+		if ctx.Err() != nil {
+			break
+		}
+		conn, err := li.Accept()
+		if err != nil {
+			warningf(ctx.App.ErrWriter, "failed to accept connection: %s", err)
+			continue
+		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			// call tunnel once per connection, the connection passed in will be closed
+			// by Tunnel.
+			if err := robotClient.Tunnel(ctx.Context, conn, dest); err != nil {
+				printf(ctx.App.Writer, "error while tunneling connection: %s", err)
+			}
+		}()
+	}
+	wg.Wait()
+	return nil
+}
+
+func (c *viamClient) robotPartTunnel(cCtx *cli.Context, args robotsPartTunnelArgs) error {
+	orgStr := args.Organization
```

Comment:
bit of a gotcha until we get to https://viam.atlassian.net/browse/RSDK-9758 (sorry!), but we should add an `ensureLoggedIn` check here.

Reply:
thanks for the reminder! completely missed that